Tasks/Common/AzureRmDeploy-common/operations/ContainerBasedDeploymentUtility.ts

```diff
@@ -22,25 +25,45 @@ export class ContainerBasedDeploymentUtility {
 
     public async deployWebAppImage(properties: any): Promise<void> {
         let imageName: string = properties["ImageName"];
-        tl.debug("Deploying an image " + imageName + " to the webapp " + this._appService.getName());
-
+        let configFilePath: string = properties["ConfigFilePath"];
+        let isMultiContainer: boolean = properties["isMultiContainer"];
         let isLinuxApp: boolean = properties["isLinuxContainerApp"];
+        let updatedConfigFilePath: string;
+        tl.debug("Deploying image " + imageName + " to the webapp " + this._appService.getName());
+
+        if(isMultiContainer) {
+            updatedConfigFilePath = this.updateImagesInConfigFile(configFilePath, imageName);
+        }
+
         tl.debug("Updating the webapp configuration.");
-        await this._updateConfigurationDetails(properties["ConfigurationSettings"], properties["StartupCommand"], imageName, isLinuxApp);
+        await this._updateConfigurationDetails(properties["ConfigurationSettings"], properties["StartupCommand"], imageName, isLinuxApp, isMultiContainer, configFilePath);
 
         tl.debug('making a restart request to app service');
         await this._appService.restart();
+
+        // uploading log file
+        if(isMultiContainer) {
```

Comment:
can we move this line along with previous `if`

Reply:
It is done.